src/basic/path-util.c

```diff
@@ -1441,3 +1441,36 @@ int path_glob_can_match(const char *pattern, const char *prefix, char **ret) {
                 *ret = NULL;
         return false;
 }
+
+const char* default_path(void) {
+#if HAVE_SPLIT_BIN
+        static int split = -1;
+        int r;
+
+        /* Check whether /usr/sbin is separate and return the appropriate $PATH.
+         * On error assume we need split bin as configured… */
+
+        if (split < 0)
+                FOREACH_STRING(path, "/usr/sbin", "/usr/local/sbin") {
+                        _cleanup_free_ char *target = NULL;
+
+                        r = chase(path, /* root= */ NULL, /* flags= */ 0, &target, NULL);
+                        if (r == -ENOENT)
+                                continue;
+                        if (r >= 0 &&
+                            path_equal_or_inode_same(target,
+                                                     streq(path, "/usr/sbin") ? "/usr/bin" : "/usr/local/bin",
+                                                     AT_NO_AUTOMOUNT))
+                                continue;
+                        if (r < 0)
+                                log_debug_errno(r, "Failed to readlink(\"/usr/sbin\"), using compat $PATH: %m");
+                        split = true;
+                        break;
```

Comment:
so, if you follow the symlink chain anyway, and then even compare inode numbers, why not just go by inode numbers directly?

i.e. replace the whole shebang with a single inode_same_at() check that just checks if the bin and sbin path point to the same inode in the end?

the chase stuff is really for cases where we need to actually canonicalize or where we need to operate on some root dir or use some of he fancy flags it provides. But here nothing of this seems necessary, two simple stat() plus comparing the result stats is entirely enough, and that's what inoe_same_at() does.

Comparing inodes directly is probably best, since it validates that the redirection really works, i.e. that you end up at the same inode. It doesn't really bother with anything else.